diff --git a/nvim/lua/usr/nvim-treesitter.lua b/nvim/lua/usr/nvim-treesitter.lua
index ffc71537274f..fd776b9bff59 100644
--- a/nvim/lua/usr/nvim-treesitter.lua
+++ b/nvim/lua/usr/nvim-treesitter.lua
@@ -78,7 +78,7 @@ require("nvim-treesitter.configs").setup({
         ["xK"] = "@class.outer",
       },
       goto_previous_start = {
-        ["xh"] = "@function.outer",
+        ["gj"] = "@function.outer",
         ["xH"] = "@class.outer",
       },
       goto_previous_end = {
diff --git a/nvim/lua/usr/which-key.lua b/nvim/lua/usr/which-key.lua
index e6f40e45cab0..6aecf0881c4c 100644
--- a/nvim/lua/usr/which-key.lua
+++ b/nvim/lua/usr/which-key.lua
@@ -119,11 +119,6 @@ wk.add({
   { "gD",        "<cmd>lua vim.lsp.buf.declaration()<cr>",               desc = "go to declaration" },
   { "gd",        "<cmd>lua vim.lsp.buf.definition()<cr>",                desc = "go to definition" },
   { "gi",        "<cmd>lua vim.lsp.buf.implementation()<cr>",            desc = "go to implementation" },
-  {
-    "gj",
-    "<cmd>lua require('barbecue.ui').navigate(-1)<cr>",
-    desc = "go to the head of function / struct",
-  },
   { "gr", "<cmd>lua vim.lsp.buf.references()<cr>",                 desc = "go to reference" },
   { "gw", "<cmd>Telescope diagnostics<cr>",                        desc = "diagnostics" },
   { "m",  group = "bookmarks" },
