diff --git a/nvim/lua/usr/init.lua b/nvim/lua/usr/init.lua
index 56f19e4c4d31..5f6fd895e4b3 100644
--- a/nvim/lua/usr/init.lua
+++ b/nvim/lua/usr/init.lua
@@ -56,14 +56,6 @@ require("bookmarks").setup({
   virt_pattern = { "*.lua", "*.md", "*.c", "*.h", "*.sh", "*.py" },
 })

-local function copy()
-  if vim.v.event.operator == "y" and vim.v.event.regname == "+" then
-    require("osc52").copy_register("+")
-  end
-end
-
-vim.api.nvim_create_autocmd("TextYankPost", { callback = copy })
-
 require("persisted").setup({
   autoload = true,
   should_save = function()
diff --git a/nvim/lua/usr/lazy.lua b/nvim/lua/usr/lazy.lua
index 91e80316b9c9..412f9f3b8448 100644
--- a/nvim/lua/usr/lazy.lua
+++ b/nvim/lua/usr/lazy.lua
@@ -153,7 +153,6 @@ require("lazy").setup({
     cmd = { "TransferInit", "TransferToggle" },
     opts = {},
   },
-  "ojroques/nvim-osc52",
   -- cppman
   {
     "madskjeldgaard/cppman.nvim",
