diff --git a/arch/x86/kernel/apic/apic.c b/arch/x86/kernel/apic/apic.c
index af49e24b46a4..6bfe5fbb18ef 100644
--- a/arch/x86/kernel/apic/apic.c
+++ b/arch/x86/kernel/apic/apic.c
@@ -1962,6 +1962,7 @@ void __init enable_IR_x2apic(void)
 	}

 	ir_stat = irq_remapping_prepare();
+	pr_info("[martins3:%s:%d] %x\n", __FUNCTION__, __LINE__, ir_stat);
 	if (ir_stat < 0 && !x2apic_supported())
 		return;

@@ -1976,8 +1977,10 @@ void __init enable_IR_x2apic(void)
 	mask_ioapic_entries();

 	/* If irq_remapping_prepare() succeeded, try to enable it */
-	if (ir_stat >= 0)
+	if (ir_stat >= 0){
 		ir_stat = irq_remapping_enable();
+		pr_info("[martins3:%s:%d] %x\n", __FUNCTION__, __LINE__, ir_stat);
+	}
 	/* ir_stat contains the remap mode or an error code */
 	try_to_enable_x2apic(ir_stat);

diff --git a/drivers/iommu/amd/init.c b/drivers/iommu/amd/init.c
index c2d80a4e5fb0..b474041efd66 100644
--- a/drivers/iommu/amd/init.c
+++ b/drivers/iommu/amd/init.c
@@ -1811,8 +1811,12 @@ static int __init init_iommu_one(struct amd_iommu *iommu, struct ivhd_header *h,
 			break;
 		}

-		if (h->efr_reg & BIT(IOMMU_EFR_XTSUP_SHIFT))
+		if (h->efr_reg & BIT(IOMMU_EFR_XTSUP_SHIFT)){
 			amd_iommu_xt_mode = IRQ_REMAP_X2APIC_MODE;
+			pr_info("[martins3:%s:%d] we have the feature \n", __FUNCTION__, __LINE__);
+		} else{
+			pr_info("[martins3:%s:%d] we don't have the feature\n", __FUNCTION__, __LINE__);
+		}

 		early_iommu_features_init(iommu, h);
