diff --git a/drivers/pci/pci-sysfs.c b/drivers/pci/pci-sysfs.c
index ab32a91f287b..5aa5c5113a23 100644
--- a/drivers/pci/pci-sysfs.c
+++ b/drivers/pci/pci-sysfs.c
@@ -29,6 +29,7 @@
 #include <linux/msi.h>
 #include <linux/of.h>
 #include <linux/aperture.h>
+#include <linux/dma-map-ops.h>
 #include "pci.h"

 static int sysfs_initialized;	/* = 0 */
@@ -373,6 +374,13 @@ static ssize_t dma_mask_bits_show(struct device *dev,
 }
 static DEVICE_ATTR_RO(dma_mask_bits);

+static ssize_t dma_ops_show(struct device *dev,
+				  struct device_attribute *attr, char *buf)
+{
+	return sprintf(buf, "%pF\n", get_dma_ops(dev));
+}
+static DEVICE_ATTR_RO(dma_ops);
+
 static ssize_t consistent_dma_mask_bits_show(struct device *dev,
 					     struct device_attribute *attr,
 					     char *buf)
@@ -608,6 +616,7 @@ static struct attribute *pci_dev_attrs[] = {
 #endif
 	&dev_attr_dma_mask_bits.attr,
 	&dev_attr_consistent_dma_mask_bits.attr,
+	&dev_attr_dma_ops.attr,
 	&dev_attr_enable.attr,
 	&dev_attr_broken_parity_status.attr,
 	&dev_attr_msi_bus.attr,
