diff --git a/tools/objtool/elf.c b/tools/objtool/elf.c
index 7ec85d567598..2099eaaa95b4 100644
--- a/tools/objtool/elf.c
+++ b/tools/objtool/elf.c
@@ -224,8 +224,11 @@ static int read_symbols(struct elf *elf)

 	symtab = find_section_by_name(elf, ".symtab");
 	if (!symtab) {
-		WARN("missing symbol table");
-		return -1;
+               /*
+                * A missing symbol table is actually possible if it's an empty
+                * .o file.  This can happen for thunk_64.o.
+                */
+               return 0;
 	}

 	symbols_nr = symtab->sh.sh_size / symtab->sh.sh_entsize;
